rootfs/etc/profile.d/_07-term-mode.sh

```
@@ -14,32 +14,87 @@
 		# Redirect output to `/dev/tty` in case we are in a subshell where output is a pipe,
 		# because this output has to go directly to the terminal.
+		local saved_state=$(stty -g)
+		[[ $GEODESIC_TRACE =~ "terminal" ]] && echo "$(tput setaf 1)* TERMINAL TRACE: Checking terminal color scheme...$(tput sgr0)" >&2
 		stty -echo
 		echo -ne '\e]10;?\a\e]11;?\a' >/dev/tty
-		IFS=: read -t 0.1 -d $'\a' x fg_rgb
-		IFS=: read -t 0.1 -d $'\a' x bg_rgb
-		stty echo
+		# Timeout of 2 was not enough when waking for sleep.
+		# The second read should be part of the first response, should not need much time at all regardless.
+		IFS=: read -s -t 1 -d $'\a' x fg_rgb
```

Shellcheck (0.10.0)

[info] read without -r will mangle backslashes.

(SC2162)